hardware/taito/src/taitoScreen.cpp

```diff
@@ -475,6 +474,39 @@ void TaitoScreen::videoRamToTaitoBuffer(sideOfScreen screenHalf)
 	}
 }
 
+int TaitoScreen::handleInput()
+{
+	// Take control of the keystate and reset/quit mutexes to avoid
+	// both threads attempting to read/write at the same time
+	pthread_mutex_lock(this->keystateLock);
+	pthread_mutex_lock(this->resetQuitLock);
+
+	int quit = 0;
+	SDL_Event event;
+	/* poll all SDL events until there are no more in the buffer. For
+	 * each event, the taitoSCreen will determine whether it cares about
+	 * the input, and will take whichever action is appropriate if it does.
+	 * If it does not care about a given event, it will be passed on to the
+	 * hardware library's update_keystates function to be dealt with there.
+	 */
+	while (SDL_PollEvent(&event))
+	{
+		if (event.type == SDL_QUIT)
+		{
+			*(this->tStruct->quit_flag) = 1;
+			quit			    = 1;
+		}
+		else
+		{
+			quit = update_keystates(this->tStruct, &event);
+		}
+	}
+
```

Comment:
A couple of thoughts, here.  Do we need update_keystates() to check for `SDL_QUIT` anymore?  If you're only passing the event into it when it's _not_ quit, we already know it can't be quit.

Also, should this be a `switch`?  Right now it's just a single if/else, but if we add more event types for `taito` to work with, a switch will make more sense; we can just call `update_keystates()` in the `default` branch.

Reply:
I agree that a switch makes more sense.
We definitely could take quit-detecting capability out of `update_keystates`. I left it this way thinking that rom libs may want to define `[Esc]` keypress behavior differently or even catch other events that cause a quit. But maybe that's overkill and/or unnecessary flexibility?

Reply:
Unnecessary flexibility given the pretty narrow focus of this particular hardware library, I mean.

Reply:
ESC is a fair point, actually.  Though we could actually catch it in `taito` if we wanted that behavior to be universal; it'd probably mean the switch would need a `goto`, though, so I guess it's a question of what's uglier.

Reply:
What about this logic:
```
while(SDL_PollEvent(&event))
{
	switch(event.type)
	{
		case SDL_keydown_event:
			switch(event.which_key_it_was)
			{
				case ESC_KEY:
					do_all_the_quit_things;
				default:
					update_keystates(event);
			};
		default:
			update_keystates(event);
	};
}
```
We just have to make sure that every default route ends in `update_keystates();`